### PR TITLE
Feature/webhooks UrbanAirship EmrEtlRunner changes

### PR DESCRIPTION
**Pull request by ninjabear**

### CODE REVIEW

**Comment by alexanderdean** [comment]
You are a hero for writing these tests @ninjabear ! Nice work

**Comment by alexanderdean** [comment]
(Once your tests have stabilized, create ticket(s) for them in your milestone and then rebase the commits onto those tickets)

**Comment by ninjabear** [comment]
Thanks @alexanderdean ! Will do

**Comment by ninjabear** [comment]
the CI results in this case are not relevant I don't think

**Comment by alexanderdean** [comment]
The Base64 implementations for the JVM are notoriously fickle across different JVMs/classpaths etc!

File: 3-enrich/emr-etl-runner/lib/snowplow-emr-etl-runner/emr_job.rb [hunk 1]
```diff
@@ -196,6 +201,10 @@ def initialize(debug, enrich, shred, elasticsearch, s3distcp, config, enrichment
           end
 
           if to_hdfs
+
+            # for ndjson/urbanairship we can group by everything, just aim for the target size
+            group_by = self.class.is_ua_ndjson(config[:collectors][:format]) ? ".*(urbanairship).*" : ".*\\.([0-9]+-[0-9]+-[0-9]+)-[0-9]+\\..*"
```
Comment (alexanderdean) [inline 3-enrich/emr-etl-runner/lib/snowplow-emr-etl-runner/emr_job.rb]: Nice

File: 3-enrich/emr-etl-runner/lib/snowplow-emr-etl-runner/s3_tasks.rb [hunk 2]
```diff
@@ -37,9 +51,17 @@ module S3Tasks
...
+      def self.build_fix_filenames(region, collector_format)
         return lambda { |basename, filepath|
+
+          if matches = collector_format.match(/^.+?\/(.+?)\//)
+            adapter = matches.captures[0]
+            if adapter == "com.urbanairship.connect"
+              return generate_short_ua_filename(filepath)
+            end
+          end
+
```
Comment (alexanderdean) [inline 3-enrich/emr-etl-runner/lib/snowplow-emr-etl-runner/s3_tasks.rb]: Nice - you made it very easy to visually verify that the existing behaviour is safely unchanged @ninjabear!

File: 3-enrich/emr-etl-runner/spec/snowplow-emr-etl-runner/runner_spec.rb [hunk 3]
```diff
@@ -41,4 +74,96 @@
...
+    expect {Runner.new args, config, enrichments, resolver}.to raise_exception(ConfigError, "collector_format 'ndjson' not supported")
+    config[:collectors][:format] = "ndjson/something"
+    expect {Runner.new args, config, enrichments, resolver}.to raise_exception(ConfigError, "collector_format 'ndjson/something' not supported")
+    config[:collectors][:format] = "ndjson/something/"
+    expect {Runner.new args, config, enrichments, resolver}.to raise_exception(ConfigError, "collector_format 'ndjson/something/' not supported")
+    config[:collectors][:format] = "ndjson/something/something"
+    Runner.new args, config, enrichments, resolver
+  end
+
 end
```
Comment (alexanderdean) [inline 3-enrich/emr-etl-runner/spec/snowplow-emr-etl-runner/runner_spec.rb]: Great new tests!

**Comment by alexanderdean** [comment]
This looks great @ninjabear - next step for me is to build this as a release candidate and get it running for some existing jobs to confirm no regressions...

**Comment by ninjabear** [comment]
Cool! Fingers crossed!

**Pull request closed without merging**